core/src/test/java/org/apache/iceberg/actions/TestBaseRewriteDataFilesAction.java

```diff
@@ -0,0 +1,185 @@
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableTestBase;
+import org.apache.iceberg.exceptions.CommitStateUnknownException;
+import org.apache.iceberg.io.FileIO;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.mockito.internal.util.collections.Sets;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.spy;
+
+@RunWith(Parameterized.class)
+public class TestBaseRewriteDataFilesAction extends TableTestBase {
+  @Parameterized.Parameters(name = "formatVersion = {0}")
+  public static Object[] parameters() {
+    return new Object[] {1, 2};
+  }
+
+  public TestBaseRewriteDataFilesAction(int formatVersion) {
+    super(formatVersion);
+  }
+
+  @Test
+  public void testReplaceCommitSuccess() throws IOException {
+    TestRewriteDataFileAction testRewriteDataFileAction = new TestRewriteDataFileAction(table);
+    DataFile dataFile = DataFiles.builder(table.spec())
+        .withFormat(FileFormat.PARQUET)
+        .withPath(Files.localOutput(temp.newFile()).location())
+        .withRecordCount(10)
+        .withFileSizeInBytes(10)
+        .build();
+
+    DataFile rewrite = DataFiles.builder(table.spec())
+        .withFormat(FileFormat.PARQUET)
+        .withPath(Files.localOutput(temp.newFile()).location())
+        .withRecordCount(10)
+        .withFileSizeInBytes(10)
+        .build();
+
+    table.newFastAppend()
+        .appendFile(dataFile)
+        .commit();
+
+    table.newFastAppend()
+        .appendFile(rewrite)
+        .commit();
+
+    testRewriteDataFileAction.replaceDataFiles(Sets.newSet(dataFile), Sets.newSet(rewrite));
+
+    Assert.assertTrue(new File(dataFile.path().toString()).exists());
+    Assert.assertTrue(new File(rewrite.path().toString()).exists());
+  }
+
+  @Test
+  public void testReplaceCommitStateUnknown() throws IOException {
+    TestRewriteDataFileAction testRewriteDataFileAction = new TestRewriteDataFileAction(table);
+    DataFile dataFile = DataFiles.builder(table.spec())
+        .withFormat(FileFormat.PARQUET)
+        .withPath(Files.localOutput(temp.newFile()).location())
+        .withFileSizeInBytes(10)
+        .withRecordCount(10)
+        .build();
+
+    DataFile rewrite = DataFiles.builder(table.spec())
+        .withFormat(FileFormat.PARQUET)
+        .withPath(Files.localOutput(temp.newFile()).location())
+        .withFileSizeInBytes(10)
+        .withRecordCount(10)
+        .build();
+
+    table.newFastAppend()
+        .appendFile(dataFile)
+        .commit();
+    table.newFastAppend()
+        .appendFile(rewrite)
+        .commit();
+
+    BaseRewriteDataFilesAction spyRewriteDataFileAction = spy(testRewriteDataFileAction);
+    doThrow(new CommitStateUnknownException(new Exception("Commit-state is unknown.")))
+        .when(spyRewriteDataFileAction)
+        .commit(any());
```

Comment:
I don't think that this actually tests the case that you're trying to. This throws an exception instead of calling `replaceDataFiles`. In order to test the catch clause you added, you'd need the table's rewrite action to throw the `CommitStateUnknownException`. You could do that with table spy that returns a rewrite spy that throws the exception from `commit`.

Reply:
@rdblue  I'm sorry I didn't catch you. I'm trying to spy spyRewriteDataFileAction, when the `commit()` function was  called in `replaceDataFiles` ,it would throw the `CommitStateUnknownException`. Then I checked the exception throw and the existence of the file. I think this can also test the case. If I have something wrong, please let me know. Thanks.

Reply:
This PR modifies `BaseRewriteDataFilesAction.replaceDataFiles`. You want to test that the update does what you expect it to. But that is not what this tests. This hijacks the call to `replaceDataFiles` not the call to `RewriteFiles.commit`, so the `RewriteFiles.commit` call never happens and the try/catch logic you updated never runs.